src/WebJobs.Script/Rpc/LanguageWorkerProcess.cs

```diff
@@ -0,0 +1,221 @@
+                          (msg.IndexOf("fail", StringComparison.OrdinalIgnoreCase) > -1) ||
+                          (msg.IndexOf("severe", StringComparison.OrdinalIgnoreCase) > -1))
+                {
+                    if (LanguageWorkerChannelUtilities.IsLanguageWorkerConsoleLog(msg))
+                    {
+                        msg = LanguageWorkerChannelUtilities.RemoveLogPrefix(msg);
+                        _workerProcessLogger?.LogError(msg);
+                    }
+                    else
+                    {
+                        _consoleLogSource?.Log(msg);
+                    }
+                    _processStdErrDataQueue = LanguageWorkerChannelUtilities.AddStdErrMessage(_processStdErrDataQueue, Sanitizer.Sanitize(msg));
+                }
+                else
+                {
+                    if (LanguageWorkerChannelUtilities.IsLanguageWorkerConsoleLog(msg))
+                    {
+                        msg = LanguageWorkerChannelUtilities.RemoveLogPrefix(msg);
+                        _workerProcessLogger?.LogInformation(msg);
+                    }
+                    else
+                    {
+                        _consoleLogSource?.Log(msg);
+                    }
+                }
+            }
+        }
+
+        private void OnProcessExited(object sender, EventArgs e)
+        {
+            if (_disposing)
+            {
+                // No action needed
+                return;
+            }
+            string exceptionMessage = string.Join(",", _processStdErrDataQueue.Where(s => !string.IsNullOrEmpty(s)));
+            try
+            {
+                if (_process.ExitCode != 0)
+                {
+                    var processExitEx = new LanguageWorkerProcessExitException($"{_process.StartInfo.FileName} exited with code {_process.ExitCode}\n {exceptionMessage}");
+                    processExitEx.ExitCode = _process.ExitCode;
+                    HandleWorkerPorcessExitError(processExitEx);
+                }
+                else
+                {
+                    _process.WaitForExit();
+                    _process.Close();
+                }
+            }
+            catch (Exception)
+            {
+                // ignore process is already disposed
+            }
+        }
+
+        private void OnOutputDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            if (e.Data != null)
+            {
+                string msg = e.Data;
+                if (LanguageWorkerChannelUtilities.IsLanguageWorkerConsoleLog(msg))
+                {
+                    msg = LanguageWorkerChannelUtilities.RemoveLogPrefix(msg);
+                    _workerProcessLogger?.LogInformation(msg);
+                }
+                else
+                {
+                    _consoleLogSource?.Log(msg);
+                }
+            }
+        }
+
+        internal void HandleWorkerPorcessExitError(LanguageWorkerProcessExitException langExc)
```

Comment:
nit: spelling here... although 'Porcess' is kind of a cool word.

Reply:
Thanks! rebased and fixed typo